R/copy-model-from.R

```diff
@@ -222,17 +222,42 @@ copy_control_stream <- function(.parent_model_path, .new_model_path, .overwrite,
   if (.update_model_file) {
 
     # read parent control stream
-    mod_str <- .parent_model_path %>% read_file()
-
-    # replace the $PROBLEM line(s)
-    new_mod_id <- get_model_id(.new_model_path)
-    mod_str <- str_replace(mod_str,
-                           "\\$PROB(.|\n)*?\\$",
-                           as.character(glue("$PROBLEM From bbr: see {new_mod_id}.yaml for details\n\n$")))
-    glue("")
+    mod_str <- tryCatch(nmrec::read_ctl(.parent_model_path), error = identity)
+
+    if(inherits(mod_str, "error")){
+      warning(
+        glue::glue(".update_model_file = TRUE was not performed. Reason: {mod_str$message}")
+      )
+      fs::file_copy(.parent_model_path, .new_model_path)
```

Comment:
I thought about adding another layer here...and checking for a specific error message (containing `$PROBLEM`), but opted not to. Let me know if doing so would make more sense though

Reply:
What you went with makes sense to me.